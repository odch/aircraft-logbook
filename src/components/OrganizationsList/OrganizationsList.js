import React from 'react'
import PropTypes from 'prop-types'
import { isLoaded } from 'react-redux-firebase'
import { FormattedMessage } from 'react-intl'
import { organization } from '../../shapes'
import withStyles from '@material-ui/core/styles/withStyles'
import Grid from '@material-ui/core/Grid'
import Card from '@material-ui/core/Card'
import CardActionArea from '@material-ui/core/CardActionArea'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'
import CreateNewFolder from '@material-ui/icons/CreateNewFolder'
import LoadingIcon from '../LoadingIcon'
import OrganizationsCreateDialog from '../OrganizationsCreateDialog'

const styles = theme => ({
  container: {
    padding: '1em',
    [theme.breakpoints.up(1000 + theme.spacing.unit * 3 * 2)]: {
      width: 1000,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  card: {
    height: 150
  },
  cardContent: {
    height: '100%'
  },
  actionArea: {
    width: '100%',
    height: '100%',
    verticalAlign: 'top'
  },
  createIcon: {
    float: 'left'
  },
  createText: {
    fontSize: '1rem',
    marginLeft: '2em'
  }
})

class OrganizationsList extends React.Component {
  componentDidMount() {
    this.props.watchOrganizations()
  }

  componentWillUnmount() {
    this.props.unwatchOrganizations()
  }

  handleCreateClick = () => {
    this.props.openCreateOrganizationDialog()
  }

  handleDialogClose = () => {
    this.props.closeCreateOrganizationDialog()
  }

  render() {
    const { organizations, classes } = this.props
    if (!isLoaded(organizations)) {
      return <LoadingIcon />
    }

    return (
      <Grid
        container
        className={classes.container}
        spacing={24}
        data-cy="organizations-list"
      >
        <Grid item key="new" sm={4} xs={12}>
          <Card className={classes.card}>
            <CardActionArea
              className={classes.actionArea}
              onClick={this.handleCreateClick}
            >
              <CardContent className={classes.cardContent}>
                <CreateNewFolder className={classes.createIcon} />
                <Typography variant="button" className={classes.createText}>
                  <FormattedMessage id="organizations.create" />
                </Typography>
              </CardContent>
            </CardActionArea>
          </Card>
        </Grid>
        {organizations.map(organization => (
          <Grid item key={organization.id} sm={4} xs={12}>
            <Card className={classes.card}>
              <CardActionArea className={classes.actionArea}>
                <CardContent className={classes.cardContent}>
                  <Typography variant="title">{organization.id}</Typography>
                </CardContent>
              </CardActionArea>
            </Card>
          </Grid>
        ))}
        <OrganizationsCreateDialog
          open={this.props.createDialogOpen}
          data={this.props.createDialogData}
          updateData={this.props.updateCreateOrganizationDialogData}
          onClose={this.handleDialogClose}
          onSubmit={this.props.createOrganization}
        />
      </Grid>
    )
  }
}

OrganizationsList.propTypes = {
  createDialogOpen: PropTypes.bool,
  createDialogData: PropTypes.object,
  organizations: PropTypes.arrayOf(organization),
  classes: PropTypes.object.isRequired,
  openCreateOrganizationDialog: PropTypes.func.isRequired,
  closeCreateOrganizationDialog: PropTypes.func.isRequired,
  updateCreateOrganizationDialogData: PropTypes.func.isRequired,
  watchOrganizations: PropTypes.func.isRequired,
  unwatchOrganizations: PropTypes.func.isRequired,
  createOrganization: PropTypes.func.isRequired
}

export default withStyles(styles)(OrganizationsList)
