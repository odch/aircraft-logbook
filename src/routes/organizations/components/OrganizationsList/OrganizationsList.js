import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import { Link } from 'react-router-dom'
import featureToggles from 'feature-toggles'
import withStyles from '@material-ui/core/styles/withStyles'
import Grid from '@material-ui/core/Grid'
import Card from '@material-ui/core/Card'
import CardActionArea from '@material-ui/core/CardActionArea'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'
import CreateNewFolder from '@material-ui/icons/CreateNewFolder'
import isLoaded from '../../../../util/isLoaded'
import { organization } from '../../../../shapes'
import LoadingIcon from '../../../../components/LoadingIcon'
import OrganizationsCreateDialog from '../OrganizationsCreateDialog'

const styles = theme => ({
  container: {
    padding: '1em',
    [theme.breakpoints.up(1000 + theme.spacing(3 * 2))]: {
      width: 1000,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  card: {
    height: 150
  },
  cardContent: {
    height: '100%'
  },
  actionArea: {
    width: '100%',
    height: '100%',
    verticalAlign: 'top'
  },
  createIcon: {
    float: 'left'
  },
  createText: {
    fontSize: '1rem'
  }
})

class OrganizationsList extends React.Component {
  handleCreateClick = () => {
    this.props.openCreateOrganizationDialog()
  }

  handleDialogClose = () => {
    this.props.closeCreateOrganizationDialog()
  }

  render() {
    const {
      createDialog,
      organizations,
      classes,
      updateCreateOrganizationDialogData,
      createOrganization
    } = this.props
    if (!isLoaded(organizations)) {
      return <LoadingIcon />
    }

    return (
      <Grid
        container
        className={classes.container}
        spacing={3}
        data-cy="organizations-list"
      >
        {featureToggles.isFeatureEnabled('organizationsManagement') && (
          <Grid item key="new" sm={4} xs={12}>
            <Card className={classes.card} data-cy="organization-create-button">
              <CardActionArea
                className={classes.actionArea}
                onClick={this.handleCreateClick}
              >
                <CardContent className={classes.cardContent}>
                  <CreateNewFolder className={classes.createIcon} />
                  <Typography variant="button" className={classes.createText}>
                    <FormattedMessage id="organizations.create" />
                  </Typography>
                </CardContent>
              </CardActionArea>
            </Card>
          </Grid>
        )}
        {organizations.map(organization => (
          <Grid item key={organization.id} sm={4} xs={12}>
            <Card className={classes.card}>
              <CardActionArea
                className={classes.actionArea}
                component={Link}
                to={`/organizations/${organization.id}`}
              >
                <CardContent className={classes.cardContent}>
                  <Typography variant="h6">{organization.id}</Typography>
                </CardContent>
              </CardActionArea>
            </Card>
          </Grid>
        ))}
        <OrganizationsCreateDialog
          open={createDialog.open}
          data={createDialog.data}
          submitted={createDialog.submitted}
          updateData={updateCreateOrganizationDialogData}
          onClose={this.handleDialogClose}
          onSubmit={createOrganization}
        />
      </Grid>
    )
  }
}

OrganizationsList.propTypes = {
  createDialog: PropTypes.shape({
    open: PropTypes.bool,
    submitted: PropTypes.bool,
    data: PropTypes.object
  }).isRequired,
  organizations: PropTypes.arrayOf(organization),
  classes: PropTypes.object.isRequired,
  openCreateOrganizationDialog: PropTypes.func.isRequired,
  closeCreateOrganizationDialog: PropTypes.func.isRequired,
  updateCreateOrganizationDialogData: PropTypes.func.isRequired,
  createOrganization: PropTypes.func.isRequired
}

export default withStyles(styles)(OrganizationsList)
