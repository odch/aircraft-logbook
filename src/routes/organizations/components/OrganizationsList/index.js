import OrganizationsList from './OrganizationsList'

export default OrganizationsList
