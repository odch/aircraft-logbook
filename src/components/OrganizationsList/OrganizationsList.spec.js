import React from 'react'
import renderer from 'react-test-renderer'
import OrganizationsList from './OrganizationsList'

describe('components', () => {
  describe('OrganizationsList', () => {
    it('renders loading icon if organizations not loaded', () => {
      const tree = renderer
        .create(<OrganizationsList loadOrganizations={() => {}} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders organizations if loaded', () => {
      const organizations = [{ id: 'org1' }, { id: 'org2' }, { id: 'org3' }]
      const tree = renderer
        .create(
          <OrganizationsList
            organizations={organizations}
            loadOrganizations={() => {}}
          />
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('calls loadOrganizations when mounted', () => {
      const loadOrganizations = jest.fn()

      renderer.create(
        <OrganizationsList loadOrganizations={loadOrganizations} />
      )

      expect(loadOrganizations).toBeCalled()
    })
  })
})
