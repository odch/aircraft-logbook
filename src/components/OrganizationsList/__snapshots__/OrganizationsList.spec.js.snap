// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components OrganizationsList renders loading icon if organizations not loaded 1`] = `
<div
  className="LoadingIcon-container-5"
>
  <div
    className="MuiCircularProgress-root-7 MuiCircularProgress-colorPrimary-10 MuiCircularProgress-indeterminate-9 LoadingIcon-icon-6"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-12"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-13 MuiCircularProgress-circleIndeterminate-15"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`components OrganizationsList renders organizations if loaded 1`] = `
<div
  className="MuiGrid-container-16 MuiGrid-spacing-xs-24-40 OrganizationsList-container-1"
  data-cy="organizations-list"
>
  <div
    className="MuiGrid-item-17 MuiGrid-grid-xs-12-56 MuiGrid-grid-sm-4-62"
  >
    <div
      className="MuiPaper-root-114 MuiPaper-elevation1-117 MuiPaper-rounded-115 MuiCard-root-113 OrganizationsList-card-2"
    >
      <button
        className="MuiButtonBase-root-144 MuiCardActionArea-root-141 OrganizationsList-actionArea-4"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <div
          className="MuiCardContent-root-147 OrganizationsList-cardContent-3"
        >
          <h2
            className="MuiTypography-root-148 MuiTypography-title-154"
          >
            org1
          </h2>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight-143"
        />
        <span
          className="MuiTouchRipple-root-174"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-17 MuiGrid-grid-xs-12-56 MuiGrid-grid-sm-4-62"
  >
    <div
      className="MuiPaper-root-114 MuiPaper-elevation1-117 MuiPaper-rounded-115 MuiCard-root-113 OrganizationsList-card-2"
    >
      <button
        className="MuiButtonBase-root-144 MuiCardActionArea-root-141 OrganizationsList-actionArea-4"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <div
          className="MuiCardContent-root-147 OrganizationsList-cardContent-3"
        >
          <h2
            className="MuiTypography-root-148 MuiTypography-title-154"
          >
            org2
          </h2>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight-143"
        />
        <span
          className="MuiTouchRipple-root-174"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-17 MuiGrid-grid-xs-12-56 MuiGrid-grid-sm-4-62"
  >
    <div
      className="MuiPaper-root-114 MuiPaper-elevation1-117 MuiPaper-rounded-115 MuiCard-root-113 OrganizationsList-card-2"
    >
      <button
        className="MuiButtonBase-root-144 MuiCardActionArea-root-141 OrganizationsList-actionArea-4"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <div
          className="MuiCardContent-root-147 OrganizationsList-cardContent-3"
        >
          <h2
            className="MuiTypography-root-148 MuiTypography-title-154"
          >
            org3
          </h2>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight-143"
        />
        <span
          className="MuiTouchRipple-root-174"
        />
      </button>
    </div>
  </div>
</div>
`;
